packages/web/src/modules/zoom/components/zoom-provider.tsx

```diff
@@ -0,0 +1,149 @@
+import React, { useEffect, useCallback, useState, ChangeEvent } from 'react';
+import { hooks, ui } from '@application';
+import { Components } from '@types';
+
+import { zoomAuthorize, generateSignature } from '../requests';
+import { ZoomContext, ZoomContextType, Role } from '../context';
+import { authorizeZoom } from '../services';
+import ZoomActivityView from './zoom-activity-view';
+
+const { useApi, useQuery } = hooks;
+const { Button, Container, Input, Row, Col, Spinner } = ui;
+
+const ZoomProvider: Components['ZoomProvider'] = ({ redirectUri }) => {
+  const [zoomContextState, setZoomContextState] = useState<ZoomContextType>({
+    userSignature: '',
+    hostUserZakToken: null,
+    meetingNumber: '',
+    username: '',
+    password: '',
+    role: null,
+  });
+
+  const { code } = useQuery<{ code?: string; path?: string }>();
+  const zoomAuthorizeApi = useApi(zoomAuthorize);
+  const zoomSignatureApi = useApi(generateSignature);
+
+  useEffect(() => {
+    if (
+      code &&
+      zoomContextState.role !== null &&
+      !zoomSignatureApi.response &&
+      !zoomSignatureApi.loading
+    ) {
+      if (zoomContextState.role === Role.Host) {
+        zoomAuthorizeApi.fetch({ code, redirectUri });
+      }
+
+      zoomSignatureApi.fetch({
+        meetingNumber: zoomContextState.meetingNumber,
+        role: zoomContextState.role as number,
+      });
+    }
+  }, [code, zoomSignatureApi, zoomAuthorizeApi, redirectUri, zoomContextState]);
+
+  useEffect(() => {
+    if (
+      !zoomSignatureApi.response?.data ||
+      !code ||
+      zoomContextState.userSignature ||
+      (!zoomAuthorizeApi.response?.data && zoomContextState.role === Role.Host)
+    ) {
+      return;
+    }
+    setZoomContextState(prevState => ({
+      ...prevState,
+      userSignature: zoomSignatureApi.response.data,
+      hostUserZakToken: zoomAuthorizeApi.response?.data,
+    }));
+  }, [zoomSignatureApi, code, zoomAuthorizeApi, zoomContextState]);
+
+  const handleInputChange = useCallback(
+    (event: ChangeEvent<HTMLInputElement>) => {
+      setZoomContextState(prevState => ({
+        ...prevState,
+        [event.target.name]:
```

Comment:
This is a bit messy. Move values in 2 variables `key` and `value`.

Can you look how do I handle form values in other components?